### PR TITLE
fix(map.jinja): GPG KEY for EPEL-8

### LINKED ISSUE #58: [BUG]  EPEL is not getting installed in rhel 8

### PR DESCRIPTION
**Pull request by itscaro**
The checksum is correct but the url is wrong.

Since I cannot find the key URL in txt format I switch to use the keys from download.fedoraproject.org instead.

Fixes #58 

### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [x] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: epel/map.jinja [hunk 1]
```diff
@@ -21,25 +21,25 @@
...
       7: {
-        'key': 'https://fedoraproject.org/static/352C64E5.txt',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-7',
         'key_hash': 'sha256=22f25ad95d5e8d371760815485dba696ea3002fc2c7f812f2c75276853387107',
         'key_name': 'RPM-GPG-KEY-EPEL-7',
         'rpm': 'http://download.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-13.noarch.rpm',
       },
       8: {
-        'key': 'https://fedoraproject.org/static/352C64E5.txt',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8',
```
Comment (itscaro) [inline epel/map.jinja:42]: The PR is to correct this GPG key, which is EPEL 7 instead of EPEL 8

File: epel/map.jinja [hunk 2]
```diff
@@ -21,25 +21,25 @@
...
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-6',
         'key_hash': 'sha256=16d35fa467c6efcee21d3aa068a02054b6a89a7a33bffa94db1fc141693d62a3',
         'key_name': 'RPM-GPG-KEY-EPEL-6',
         'rpm': 'http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm',
       },
       7: {
-        'key': 'https://fedoraproject.org/static/352C64E5.txt',
-        'key_hash': 'sha256=22f25ad95d5e8d371760815485dba696ea3002fc2c7f812f2c75276853387107',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-7',
+        'key_hash': 'sha256=028b9accc59bab1d21f2f3f544df5469910581e728a64fd8c411a725a82300c2',
```
Comment (itscaro) [inline epel/map.jinja:37]:
```
gpg --with-fingerprint 352C64E5.txt
pub  4096R/352C64E5 2013-12-16 Fedora EPEL (7) <epel@fedoraproject.org>
      Key fingerprint = 91E9 7D7C 4A5E 96F1 7F3E  888F 6A2F AEA2 352C 64E5
```
```
gpg --with-fingerprint RPM-GPG-KEY-EPEL-7
pub  4096R/352C64E5 2013-12-16 Fedora EPEL (7) <epel@fedoraproject.org>
      Key fingerprint = 91E9 7D7C 4A5E 96F1 7F3E  888F 6A2F AEA2 352C 64E5
```

**Comment by myii** [comment]
Thanks for working through the links, @itscaro.  I'd like to get a review from @ixs who had a big part in setting up the original structure of this file.  It make take some time because of the holidays but if you don't hear from us, please feel free to ping me.

**Comment by ixs** [comment]
Thank you @myii. I had a quick look:

The change looks generally good and coincidentally is very similar to what I had locally but forgot to issue a PR for. 🤦🏻‍♂️
The SHA256SUM also looks good and is due to different representation of the GPG key on the fedoraproject site vs. the download site.

There's just one thing I'd like you to fix before approving. Please change the links from https://download.fedoraproject.org/ to https://dl.fedoraproject.org/.

The downloads vhost is serving a redirect to the mirror network. The dl vhost is the canonical master server. For something like the GPG signing key I'd prefer the canonical master without any mirrors in between that might break things in unexpected way.

File: epel/map.jinja [hunk 3]
```diff
@@ -21,25 +21,25 @@
     default='2'),
     'RedHat': salt['grains.filter_by']({
       5: {
-        'key': 'https://fedoraproject.org/static/A4D647E9.txt',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-5',
```
Comment (ixs) [inline epel/map.jinja:24]: dl.fedoraproject.org please.

File: epel/map.jinja [hunk 4]
```diff
@@ -21,25 +21,25 @@
...
         'key_hash': 'sha256=664c06f579d914f2cf25d05d4d581b8ddec77cae4f72f4020c3a9264b9d5ee71',
         'key_name': 'RPM-GPG-KEY-EPEL-5',
         'rpm': 'http://download.fedoraproject.org/pub/archive/epel/5/i386/epel-release-5-4.noarch.rpm',
       },
       6: {
-        'key': 'https://fedoraproject.org/static/0608B895.txt',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-6',
```
Comment (ixs) [inline epel/map.jinja:30]: same. dl.fedoraproject.org please.

File: epel/map.jinja [hunk 5]
```diff
@@ -21,25 +21,25 @@
...
         'key_hash': 'sha256=16d35fa467c6efcee21d3aa068a02054b6a89a7a33bffa94db1fc141693d62a3',
         'key_name': 'RPM-GPG-KEY-EPEL-6',
         'rpm': 'http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm',
       },
       7: {
-        'key': 'https://fedoraproject.org/static/352C64E5.txt',
-        'key_hash': 'sha256=22f25ad95d5e8d371760815485dba696ea3002fc2c7f812f2c75276853387107',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-7',
```
Comment (ixs) [inline epel/map.jinja:36]: same. dl.fedoraproject.org please.

File: epel/map.jinja [hunk 6]
```diff
@@ -21,25 +21,25 @@
...
+        'key_hash': 'sha256=028b9accc59bab1d21f2f3f544df5469910581e728a64fd8c411a725a82300c2',
         'key_name': 'RPM-GPG-KEY-EPEL-7',
         'rpm': 'http://download.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-13.noarch.rpm',
       },
       8: {
-        'key': 'https://fedoraproject.org/static/352C64E5.txt',
+        'key': 'https://download.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8',
```
Comment (ixs) [inline epel/map.jinja:42]: same. dl.fedoraproject.org please.

**Review by ixs — CHANGES_REQUESTED** [review]

**Comment by myii** [comment]
Really appreciate the prompt review, @ixs.  Excellent feedback, too.

**Review by ixs — APPROVED** [review]
@itscaro Thank you very much. Looks good.

**Comment by itscaro** [comment]
@ixs Thank you for the feedback, I switch to dl.fedoraproject.org :-)

Happy holidays!

**Comment by ixs** [comment]
@itscaro merry christmas or whatever floats your sled. 😂

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
A nice, tidy PR -- thanks guys!  Merged.

**Pull request merged**